### DISCUSSION TITLE
Device name vs Friendly Name 1

### DISCUSSION THREAD

**Discussion by NsinghP** [arendst/Tasmota#11687] [Q&A] [answered]
Hello, 

Can Friendly_Name feature be modified easily? It is rather inconvenient to not be able to set a device name that is not dependent on Friendly Name 1. As you can see in screenshot below I have set the Friendly Name 1 to "First Room Switch" as a result Device name has automatically become "First Room Switch". I would rather have the Device name as "First Room" and Friendly name 1 as "First Room Switch 1" but I can't do that because it will change Device name to ""First Room Switch 1".

Is there any workaround?
Is there a reason, that I don't understand, which prevents implementation of this feature?

It will be nice to be able to name the device independently of Switch Names.

Regards,

Navi

![Screenshot (86)](https://user-images.githubusercontent.com/69470381/114414337-58c6f680-9bcc-11eb-95ff-d133fa182343.png)

**Comment by ascillato** [comment, marked as answer]
Please, use command `devicename`

Check Tasmota Docs at https://tasmota.github.io/docs/Commands/#management for more information

**Reply by NsinghP** [reply]
Thanks, Is it possible to use this command in user_config_override.h and make custom device name part of the bin file? or is it better to use autoexec.bat file to execute this command?

**Reply by ascillato2** [reply]
device name can not be defined in user_config_override.h. It is better to use autoexec.bat

**Reply by barbudor** [reply]
DeviceName can be different than FriendlyName1
It is just that originally there was only FriendlyName, then FriendlyName1..4 and later on DeviceName was introduced to differentiate the device and each switch/relay
But user_config_override hasn't been changed. Only FriendlyName(1) can be preset in the firmware, and DeviceName will be by default initialized to this value.

Let me see if we can improve that

**Reply by barbudor** [reply]
@NsinghP 
Now you can `#define DEVICE_NAME` in `user_config_override.h`
https://github.com/arendst/Tasmota/pull/11694

**Reply by NsinghP** [reply]
Thanks barbudor, 

That saves me from making and uploading autoexec.bat file. (Though i am still reading up on rules documentation to figure that out). Had you not enabled user_config_override.h ability to define device name, my next question would have been how to integrate autoexec.bat file with .bin image so that both can be burned to flash at the same time. Maybe i will ask that question later on when i have mastered the autoexec.bat file structure.

Regards,

Navi
